### PR TITLE
[manifest] #2  add a module to abstract operations to repository

### PR DESCRIPTION
**Pull request by PengTian0**
Add a module RepositoryOperator to abstract operations to repository.

The proposal of manifest build in rackhd release  is [here](https://groups.google.com/forum/#!topic/rackhd/v3LpRuQJvQg).

This PR depends on https://github.com/RackHD/on-tools/pull/25

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000

**Comment by PengTian0** [comment]
@RackHD/corecommitters

File: manifest-build-tools/lib/RepositoryOperator.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+            error = False
+            for name in results.keys():
+                error |= self.print_command_summary(name, results)
+
+            if error:
+                raise RuntimeError("Failed to clone repositories")
+
+    
+    def clone_repo(self, repo_url, dest_dir, repo_commit="HEAD"):
```
Comment (panpan0000) [inline manifest-build-tools/lib/RepositoryOperator.py]: What's the difference than ```class RepoCloner```?

**Comment by panpan0000** [comment]
It's better to give a quick **example** in review description, about how to use the 2 classes.

File: manifest-build-tools/lib/RepositoryOperator.py (hunk 1 above)
Comment (PengTian0) [inline manifest-build-tools/lib/RepositoryOperator.py]: The class RepoCloner support add several tasks and clone repositories in parallel.
An example of the RepoCloner is the function clone_repo_list

**Comment by panpan0000** [comment]
Hi, @PengTian0 , Although I can find RepoCloner example in RepoOperator() class.
but I still insist to add short usage example in RepoCloner()'s comments (Line 12).
so the class itself is a completed and pretty class.

**Comment by PengTian0** [comment]
OKay!
I will submit another commit to add the usage of RepoCloner~

**Comment by panpan0000** [comment]
:+1:

**Pull request merged**